generate-spec.php

```diff
@@ -83,10 +87,14 @@
 	if ($xml === false) {
 		Logger::panic('appinfo', 'info.xml file at ' . $infoXMLPath . ' is not parsable');
 	}
+	if (!$xml->namespace) {
+		Logger::panic('appinfo', 'A <namespace/> must be set.');
+	}
```

Comment:
Seems like this is allowed, so the namespace must be assembled manually from the app ID here.